### PR TITLE
refactor: remove lodash & use native javascript instead

### PR DESCRIPTION
**Pull request by xxczaki**
## Introduction

This might be a controversial change, but I'm prepared it might not get merged.

## Description

This pull request removes lodash and replaces all of it's methods with native JavaScript methods.

> It's always better to use native JavaScript, than rely on heavy libraries.

I used the suggested lodash replacements from the ["You don't (may not) need Lodash/Underscore" repository](https://github.com/you-dont-need/You-Dont-Need-Lodash-Underscore).

If you have any suggestions, let me know down below.

### CODE REVIEW

File: core/build-readme.js [hunk 1]
```diff
@@ -13,6 +12,9 @@ import qsm from 'qsm'
 
 import build from './build'
 
+const isEmpty = obj =>
+  [Object, Array].includes((obj || {}).constructor) && !Object.entries(obj || {}).length
```
Comment (wooorm) [inline core/build-readme.js]: This one liner is really hard to read for me

File: server.js [hunk 2]
```diff
@@ -12,7 +11,7 @@ const routes = require('./routes')
 
 const { INTERNAL_NEXT_PATHS, NODE_ENV, PORT, GITHUB_TOKEN } = require('./constants')
 
-if (isNil(GITHUB_TOKEN)) {
+if (GITHUB_TOKEN === null || undefined) {
```
Comment (wooorm) [inline server.js]:
```suggestion
if (GITHUB_TOKEN === null || GITHUB_TOKEN === undefined) {
```

File: core/build-readme.js (hunk 1 above)
Comment (xxczaki) [reply core/build-readme.js]: @wooorm You are right. Now it should be more readable :smile:

**Comment by wooorm** [comment]
I share @Kikobeats conserns btw!

**Comment by xxczaki** [comment]
@Kikobeats Sounds good to me. Let's use lodash for the `isEmpty` and `get` methods. I will make changes in a few hours, when i get home :smile:

**Comment by xxczaki** [comment]
@Kikobeats Done.

File: core/github/index.js [hunk 3]
```diff
@@ -14,9 +13,7 @@ const cases = (name, ext) => [
 
 const { GITHUB_TOKEN } = process.env || {}
 
-// based on https://github.com/github/markup#markups
```
Comment (Kikobeats) [inline core/github/index.js]: don't remove this comment please

File: server.js [hunk 4]
```diff
@@ -12,7 +11,7 @@ const routes = require('./routes')
 
 const { INTERNAL_NEXT_PATHS, NODE_ENV, PORT, GITHUB_TOKEN } = require('./constants')
 
-if (isNil(GITHUB_TOKEN)) {
+if (GITHUB_TOKEN === null || GITHUB_TOKEN === undefined) {
```
Comment (Kikobeats) [inline server.js]: that's isNil implementation, use it

File: core/build-readme.js [hunk 5]
```diff
@@ -80,7 +80,7 @@ const withExternalIcon = $ => {
 }
 
 const createWithRelativeLinks = ({ isMarkdownPath }) => ($, { ref }) => {
-  forEach(TAGS, (htmlTags, propName) => {
+  Array.from(TAGS).forEach((htmlTags, propName) => {
```
Comment (Kikobeats) [inline core/build-readme.js]: is this introducing a bug?

```
> require('html-urls').TAGS
{ background: [ 'body' ],
  cite: [ 'blockquote', 'del', 'ins', 'q' ],
  data: [ 'object' ],
  href: [ 'a', 'area', 'embed', 'link' ],
  icon: [ 'command' ],
  longdesc: [ 'frame', 'iframe' ],
  manifest: [ 'html' ],
  poster: [ 'video' ],
  pluginspage: [ 'embed' ],
  pluginurl: [ 'embed' ],
  src:
   [ 'audio',
     'embed',
     'frame',
     'iframe',
     'img',
     'input',
     'script',
     'source',
     'track',
     'video' ] }
> Array.from(require('html-urls').TAGS)
[]
```

File: core/github/index.js (hunk 3 above)
Comment (xxczaki) [reply core/github/index.js]: My mistake - sorry!

File: core/build-readme.js (hunk 5 above)
Comment (xxczaki) [reply core/build-readme.js]: I'm not sure, let's just use the lodash implementation - that's the safest option.

File: server.js (hunk 4 above)
Comment (xxczaki) [reply server.js]: Done

**Pull request merged**